games/minimal/mods/default/init.lua

```diff
@@ -1158,7 +1158,9 @@ minetest.register_node("default:chest", {
 		meta:set_string("formspec",
 				"size[8,9]"..
 				"list[current_name;main;0,0;8,4;]"..
-				"list[current_player;main;0,5;8,4;]")
+				"list[current_player;main;0,5;8,4;]" ..
+				"listring[current_name;main]" ..
+				"listring[current_player;main]")
```

Comment:
You basically create a ring where `current_name;main` is followed by `current_player;main` which is again followed by `current_name;main`, like in a ring.

In this example, it means shift-clicking from a chest puts item into the inventory, and vice versa.

You can have more complex examples too, see furnace, where the output slot maps into the inventory, but the inventory maps into the input slot, which itself again maps into the inventory. This is what I've meant with "First occurence counts".